### PR TITLE
DROOLS-3469 DMN xml NS definition on non-root xml elements

### PR DESCRIPTION
**Pull request by tarilabs**
/cc @manstis @jomarko @danielezonca 

Linked PR also : https://github.com/kiegroup/drools/pull/2205

The linked PR fixes the issue.
This PR additionally avoid the issue beforehand.

### CODE REVIEW

**Comment by tarilabs** [comment]
Before example:
```
<?xml version='1.0' encoding='UTF-8'?>
<dmn:definitions xmlns:dmn="http://www.omg.org/spec/DMN/20180521/MODEL/" xmlns="https://github.com/kiegroup/drools/kie-dmn/_41A586D4-CEE9-420F-9289-7E0249B2EA34" xmlns:feel="http://www.omg.org/spec/DMN/20180521/FEEL/" xmlns:kie="http://www.drools.org/kie/dmn/1.2" id="_EA28DEE4-22A9-44DC-B468-B6012DCB1C86" name="dmn1" expressionLanguage="http://www.omg.org/spec/DMN/20180521/FEEL/" typeLanguage="http://www.omg.org/spec/DMN/20180521/FEEL/" namespace="https://github.com/kiegroup/drools/kie-dmn/_41A586D4-CEE9-420F-9289-7E0249B2EA34">
  <dmn:extensionElements/>
  <dmn:decision id="_FDFD4EBA-2A7A-4BE9-B09A-4D0705BACC9A" name="is it cold?">
    <dmn:variable id="_88A0CF06-5A68-45F4-95FD-B206627C437C" name="is it cold?" typeRef="string"/>
    <dmn:informationRequirement id="_F5F1FB04-9839-4072-9528-786443DD76E3">
      <dmn:requiredInput href="#_38A652E8-F7ED-4E70-9C7C-502FA20BFC19"/>
    </dmn:informationRequirement>
    <dmn:decisionTable id="_DB550D9B-4FC5-4FA2-92AB-52DC2E18E0DA" hitPolicy="ANY" preferredOrientation="Rule-as-Row">
      <dmn:input id="_A5E94A1A-C183-4639-8D11-86B5303EBF3B">
        <dmn:inputExpression id="_F25592F9-57A3-4D9D-B38F-C6AF96BAE83A" typeRef="number">
          <dmn:text>temperature</dmn:text>
        </dmn:inputExpression>
      </dmn:input>
      <dmn:output id="_AB07B6EE-CFEF-4991-BA94-0627F153ABAB" typeRef="string"/>
      <dmn:rule id="_AF3AAF21-984B-4372-8718-EFCB85E48A32">
        <dmn:inputEntry id="_C1A05A14-86A2-4A27-AFD7-755FD7D46367">
          <dmn:text>&lt;= 0</dmn:text>
        </dmn:inputEntry>
        <dmn:outputEntry id="_26832DFA-9C24-4A41-B411-B681E40947C5">
          <dmn:text>"cold"</dmn:text>
        </dmn:outputEntry>
      </dmn:rule>
      <dmn:rule id="_554465C4-FED0-436F-87A5-483D47B1779C">
        <dmn:inputEntry id="_E5A9B5F8-B3AC-49ED-89D3-F08A2B04839E">
          <dmn:text>&gt; 0</dmn:text>
        </dmn:inputEntry>
        <dmn:outputEntry id="_848D0FF0-EC90-4A1E-A984-69D79B6A0F5F">
          <dmn:text>"hot"</dmn:text>
        </dmn:outputEntry>
      </dmn:rule>
    </dmn:decisionTable>
  </dmn:decision>
  <dmn:inputData id="_38A652E8-F7ED-4E70-9C7C-502FA20BFC19" name="temperature">
    <dmn:variable id="_543E5A1E-3ED3-43E5-B2EC-1425B23EC66F" name="temperature" typeRef="number"/>
  </dmn:inputData>
  <dmndi:DMNDI xmlns:dmndi="http://www.omg.org/spec/DMN/20180521/DMNDI/">
    <dmndi:DMNDiagram>
      <dmndi:DMNShape id="dmnshape-_FDFD4EBA-2A7A-4BE9-B09A-4D0705BACC9A" dmnElementRef="_FDFD4EBA-2A7A-4BE9-B09A-4D0705BACC9A" isCollapsed="false">
        <dmndi:DMNStyle>
          <dmndi:FillColor red="255" green="255" blue="255"/>
          <dmndi:StrokeColor red="0" green="0" blue="0"/>
          <dmndi:FontColor red="0" green="0" blue="0"/>
        </dmndi:DMNStyle>
        <dc:Bounds xmlns:dc="http://www.omg.org/spec/DMN/20180521/DC/" x="424" y="130" width="200" height="122"/>
        <dmndi:DMNLabel/>
      </dmndi:DMNShape>
      <dmndi:DMNShape id="dmnshape-_38A652E8-F7ED-4E70-9C7C-502FA20BFC19" dmnElementRef="_38A652E8-F7ED-4E70-9C7C-502FA20BFC19" isCollapsed="false">
        <dmndi:DMNStyle>
          <dmndi:FillColor red="255" green="255" blue="255"/>
          <dmndi:StrokeColor red="0" green="0" blue="0"/>
          <dmndi:FontColor red="0" green="0" blue="0"/>
        </dmndi:DMNStyle>
        <dc:Bounds xmlns:dc="http://www.omg.org/spec/DMN/20180521/DC/" x="216" y="165" width="100" height="50"/>
        <dmndi:DMNLabel/>
      </dmndi:DMNShape>
      <dmndi:DMNEdge id="dmnedge-_F5F1FB04-9839-4072-9528-786443DD76E3" dmnElementRef="_F5F1FB04-9839-4072-9528-786443DD76E3">
        <di:waypoint xmlns:di="http://www.omg.org/spec/DMN/20180521/DI/" x="266" y="190"/>
        <di:waypoint xmlns:di="http://www.omg.org/spec/DMN/20180521/DI/" x="424" y="191"/>
      </dmndi:DMNEdge>
    </dmndi:DMNDiagram>
  </dmndi:DMNDI>
</dmn:definitions>
```

We can notice NS for dmndi, di, dc is not performed on the root xml element.

After example:
```
[INFO] <?xml version="1.0" ?>
[INFO] <dmn:definitions xmlns:dmn="http://www.omg.org/spec/DMN/20180521/MODEL/" xmlns="https://github.com/kiegroup/drools/kie-dmn/_6B89E392-FC90-4B78-B4E6-5B79ADEF1F77" xmlns:di="http://www.omg.org/spec/DMN/20180521/DI/" xmlns:kie="http://www.drools.org/kie/dmn/1.2" xmlns:feel="http://www.omg.org/spec/DMN/20180521/FEEL/" xmlns:dmndi="http://www.omg.org/spec/DMN/20180521/DMNDI/" xmlns:dc="http://www.omg.org/spec/DMN/20180521/DC/" id="_656A75B8-5675-432B-8B9B-F2C4505FF15A" name="_8B97856A-3DBE-4FB2-BA25-10462A3F2E31" expressionLanguage="http://www.omg.org/spec/DMN/20180521/FEEL/" typeLanguage="http://www.omg.org/spec/DMN/20180521/FEEL/" namespace="https://github.com/kiegroup/drools/kie-dmn/_6B89E392-FC90-4B78-B4E6-5B79ADEF1F77">
[INFO]   <dmn:extensionElements></dmn:extensionElements>
[INFO]   <dmn:decision id="_4AA167FA-240B-40AB-8AD9-2B460D7A01E1" name="Decision-1">
[INFO]     <dmn:variable id="_428D769F-F5E7-41E2-9AF4-987FAC499422" name="Decision-1"></dmn:variable>
[INFO]     <dmn:literalExpression></dmn:literalExpression>
[INFO]   </dmn:decision>
[INFO]   <dmn:decision id="_362CBB29-604E-4914-8807-9726689F345A" name="Decision-2">
[INFO]     <dmn:variable id="_9C2EC447-AA51-45A6-8CE5-C8A9011B8A1F" name="Decision-2"></dmn:variable>
[INFO]     <dmn:informationRequirement id="_F56F69EA-DBB4-4D45-BAEF-0E4D5C64284B">
[INFO]       <dmn:requiredDecision href="#_4AA167FA-240B-40AB-8AD9-2B460D7A01E1"></dmn:requiredDecision>
[INFO]     </dmn:informationRequirement>
[INFO]     <dmn:literalExpression></dmn:literalExpression>
[INFO]   </dmn:decision>
[INFO]   <dmndi:DMNDI>
[INFO]     <dmndi:DMNDiagram>
[INFO]       <dmndi:DMNShape id="dmnshape-_362CBB29-604E-4914-8807-9726689F345A" dmnElementRef="_362CBB29-604E-4914-8807-9726689F345A" isCollapsed="false">
[INFO]         <dmndi:DMNStyle>
[INFO]           <dmndi:FillColor red="255" green="255" blue="255"></dmndi:FillColor>
[INFO]           <dmndi:StrokeColor red="0" green="0" blue="0"></dmndi:StrokeColor>
[INFO]           <dmndi:FontColor red="0" green="0" blue="0"></dmndi:FontColor>
[INFO]         </dmndi:DMNStyle>
[INFO]         <dc:Bounds x="283" y="102" width="100" height="50"></dc:Bounds>
[INFO]         <dmndi:DMNLabel></dmndi:DMNLabel>
[INFO]       </dmndi:DMNShape>
[INFO]       <dmndi:DMNShape id="dmnshape-_4AA167FA-240B-40AB-8AD9-2B460D7A01E1" dmnElementRef="_4AA167FA-240B-40AB-8AD9-2B460D7A01E1" isCollapsed="false">
[INFO]         <dmndi:DMNStyle>
[INFO]           <dmndi:FillColor red="255" green="255" blue="255"></dmndi:FillColor>
[INFO]           <dmndi:StrokeColor red="0" green="0" blue="0"></dmndi:StrokeColor>
[INFO]           <dmndi:FontColor red="0" green="0" blue="0"></dmndi:FontColor>
[INFO]         </dmndi:DMNStyle>
[INFO]         <dc:Bounds x="103" y="102" width="100" height="50"></dc:Bounds>
[INFO]         <dmndi:DMNLabel></dmndi:DMNLabel>
[INFO]       </dmndi:DMNShape>
[INFO]       <dmndi:DMNEdge id="dmnedge-_F56F69EA-DBB4-4D45-BAEF-0E4D5C64284B" dmnElementRef="_F56F69EA-DBB4-4D45-BAEF-0E4D5C64284B">
[INFO]         <di:waypoint x="153" y="127"></di:waypoint>
[INFO]         <di:waypoint x="333" y="127"></di:waypoint>
[INFO]       </dmndi:DMNEdge>
[INFO]     </dmndi:DMNDiagram>
[INFO]   </dmndi:DMNDI>
[INFO] </dmn:definitions>
```

We can notice NS for dmndi, di, dc is defined on the root xml element.

File: kie-wb-common-dmn/kie-wb-common-dmn-api/src/main/java/org/kie/workbench/common/dmn/api/definition/v1_1/DMNModelInstrumentedBase.java [hunk 1]
```diff
@@ -39,7 +39,10 @@
         FEEL("feel", org.kie.dmn.model.v1_2.KieDMNModelInstrumentedBase.URI_FEEL),
         DMN("dmn", org.kie.dmn.model.v1_2.KieDMNModelInstrumentedBase.URI_DMN),
         KIE("kie", org.kie.dmn.model.v1_2.KieDMNModelInstrumentedBase.URI_KIE),
-        DEFAULT(QName.DEFAULT_NS_PREFIX, "https://github.com/kiegroup/drools/kie-dmn/");
+        DEFAULT(QName.DEFAULT_NS_PREFIX, "https://github.com/kiegroup/drools/kie-dmn/"),
+        DMNDI("dmndi", org.kie.dmn.model.v1_2.KieDMNModelInstrumentedBase.URI_DMNDI),
```
Comment (manstis) [inline kie-wb-common-dmn/kie-wb-common-dmn-api/src/main/java/org/kie/workbench/common/dmn/api/definition/v1_1/DMNModelInstrumentedBase.java]: Could you please also amend [DMNDiagramFactoryImplTest](https://github.com/kiegroup/kie-wb-common/blob/master/kie-wb-common-dmn/kie-wb-common-dmn-api/src/test/java/org/kie/workbench/common/dmn/api/factory/DMNDiagramFactoryImplTest.java).

**Comment by tarilabs** [comment]
@manstis Yes, but I tested on the linked PR the _actual_ problem and the fix there directly. If there are additional issues I'm sure will be arised later too. This PR avoid the issue (even if the fix is fixed, in order to have a more consistent way to define namespaces while serializing the XML)

**Comment by tarilabs** [comment]
Jenkins test again

**Pull request merged**